src/Persistence/Fake/FakeResetPasswordInternalRepository.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace SymfonyCasts\Bundle\ResetPassword\Persistence\Fake;
+
+use SymfonyCasts\Bundle\ResetPassword\Exception\FakeRepositoryException;
+use SymfonyCasts\Bundle\ResetPassword\Model\ResetPasswordRequestInterface;
+use SymfonyCasts\Bundle\ResetPassword\Persistence\ResetPasswordRequestRepositoryInterface;
+
+/**
+ * Class is only used as a placeholder and it's signature should be replaced in reset_password.yaml.
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver   <ryan@symfonycasts.com>
+ *
+ * @internal
+ */
+final class FakeResetPasswordInternalRepository implements ResetPasswordRequestRepositoryInterface
+{
+    public function createResetPasswordRequest(object $user, \DateTimeInterface $expiresAt, string $selector, string $hashedToken): ResetPasswordRequestInterface
+    {
+        $this->throwException();
+    }
+
+    public function getUserIdentifier(object $user): string
+    {
+        $this->throwException();
+    }
+
+    public function persistResetPasswordRequest(ResetPasswordRequestInterface $resetPasswordRequest): void
+    {
+        $this->throwException();
+    }
+
+    public function findResetPasswordRequest(string $selector): ?ResetPasswordRequestInterface
+    {
+        $this->throwException();
+    }
+
+    public function getMostRecentNonExpiredRequestDate(object $user): ?\DateTimeInterface
+    {
+        $this->throwException();
+    }
+
+    public function removeResetPasswordRequest(ResetPasswordRequestInterface $resetPasswordRequest): void
+    {
+        $this->throwException();
+    }
+
+    public function removeExpiredResetPasswordRequests(): int
+    {
+        $this->throwException();
+    }
+
+    private function throwException(): void
+    {
+        throw new FakeRepositoryException();
```

Comment:
I'd just put this up in each method - will read nice and simple

Reply:
done